### PR TITLE
Allow config.*.y*ml files to override config.yaml, fixes #1410

### LINKED ISSUE #1410: Allow overriding config.yaml so that .ddev config can be meaningfully committed to Git
**Is your feature request related to a problem? Please describe.**
If I commit the DDEV directory to Git with, say, `webcache_enabled: true`, then my colleague on Linux will have to turn it off (there is no point in using it on Linux). Same with the upcoming NFS feature.

Ideally, I could specify:

````yaml
additional_config:
  - ./local.config.yaml
````

and ignore the local config in Git. Then in my local config I could turn on webcache or NFS, or set up additional domains that not everyone needs, etc.

**Describe the solution you'd like**
````yaml
additional_config:
  - ./local.config.yaml
````

Additional config is read after `.ddev/config.yaml`.

**Describe alternatives you've considered**
A private Git branch?
Just letting the config file be dirty on my local?
Not committing `.ddev`?

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by rfay**
## The Problem/Issue/Bug:

OP #1410 requests config.yaml overrides. There are lots of different opinions in the thread, so a very simple approach is used here.

## How this PR Solves The Problem:

* .ddev/config.yaml is loaded first
* After that, override files in .ddev/config.*.y*ml are loaded, in alphabetical/glob order. 

Users can choose the workflow they want for these and can choose how to manage them in git.

The simplest possible technique could be to use a `config.local.yaml` for things that should not be shared with the team.

## TODO:

- [x] Add `ddev debug configyaml` to show the resultant config.yaml totally loaded, and also make sure it shows what files are being loaded.
- [x] Make sure that `ddev config` and anything else that writes config.yaml *only* writes stuff that was in the config.yaml, not in the add-ons. Probably need a test for this.
- [x] Add docs on how this works

## Manual Testing Instructions:

* Add a `.ddev/config.local.yaml` with custom configuration. `ddev start` and inspect the results.
* Try this with multiple files.
* Test multiple variants of `ddev config` (which writes out config.yaml) to make sure that config.*.yaml values cannot sneak into the written config.yaml.
* Verify the correctness of the output of `ddev debug configyaml`
* Verify the correctness of new docs

## Automated Testing Overview:

- [x] Test this basic behavior
- [x] Test that config.*.yaml are properly ordered by the glob on all test environments.

## Related Issue Link(s):

OP #1410

### CODE REVIEW

**Comment by rfay** [comment]
Artifacts for this version for testing are at https://circleci.com/gh/drud/ddev/11014#artifacts/containers/0

**Comment by rfay** [comment]
As I think about this there's probably an untested flaw; `ddev config` will probably write out the fully-concatenated version of the config...

**Comment by Lefaux** [comment]
I strongly recommend to not implement this change as is.
glob() orders vary between OSs which can lead to unexpected results which are hard, if not impossible to debug.

Manual sorting could help here, so an alphabetical sorting in go could solve that problem.

File: pkg/ddevapp/config.go [hunk 1]
```diff
@@ -195,25 +195,27 @@ func (app *DdevApp) WriteConfig() error {
...
-		return fmt.Errorf("invalid configuration in %s: %v", app.ConfigPath, err)
+		return fmt.Errorf("unable to load config file %s: %v", app.ConfigPath, err)
 	}
-
 	app.DBImage = "" // DBImage will be set below
 
-	// ReadConfig config values from file.
-	err = yaml.Unmarshal(source, app)
+	// Load config.*.y*ml after in glob order
+	glob := filepath.Join(filepath.Dir(app.ConfigPath), "config.*.y*ml")
```
Comment (unn) [inline pkg/ddevapp/config.go]: A `config.*.y?ml` makes more sense to me.

**Comment by rfay** [comment]
Thanks @Lefaux - This isn't ready for commit by any means, and I certainly agree that the results of the glob have to be consistent, and we need to test for this. I do note that glob is not an operating system function, but (normally) a shell function, and that here we're using golang's globbing, not the shell's. Sorting is easy if it's not irrelevant, but it seems to me that [the golang glob code](https://golang.org/src/path/filepath/match.go?s=5600:5655#L305) is committed to lexicographical order.

File: pkg/ddevapp/config.go (hunk 1 above)
Comment (rfay) [reply pkg/ddevapp/config.go]: Thanks for taking a look @unn - but y?ml doesn't match config.*.yml, which is the intent here. It seems [glob has no concept of 0 or 1 characters](https://en.wikipedia.org/wiki/Glob_(programming)) like some versions of regex do. Of course we could use regex here.

Comment (unn) [reply pkg/ddevapp/config.go]: I tried! On the upside we can say we support `yaaaaaaaml` files.

Comment (rfay) [reply pkg/ddevapp/config.go]: And config.yWHATDOYOUMEANml!

Comment (rfay) [reply pkg/ddevapp/config.go]: Oh, I guess we do config.what.in.the.hell.is.in.this.yANIml!

**Comment by rfay** [comment]
Latest artifacts for testing are at https://circleci.com/gh/drud/ddev/11174#artifacts/containers/0 - Thanks to all who are willing to take it for a spin!

**Comment by alesrebec** [comment]
Just for reference... because we used this structure `config.*.yaml` we can experiment more with file naming.
Just tried using ddev v.1.7.1 with config override named `config.os.linux.local.yaml` and it works as it should.
I only included those two lines and they were both used as overrides from original config.yaml.

```
APIVersion: v1.7.1
xdebug_enabled: true
```

**Comment by rfay** [comment]
Thanks for the feedback! Glad it's going well.

**Pull request merged**